demos/appyx-interactions/desktop/src/desktopMain/kotlin/com/bumble/appyx/interactions/widgets/ui/WidgetsStack3D.kt

```diff
@@ -0,0 +1,84 @@
+package com.bumble.appyx.interactions.widgets.ui
+
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.DpOffset
+import androidx.compose.ui.unit.dp
+import com.bumble.appyx.components.spotlight.SpotlightModel
+import com.bumble.appyx.interactions.core.ui.context.UiContext
+import com.bumble.appyx.interactions.core.ui.property.impl.Alpha
+import com.bumble.appyx.interactions.core.ui.property.impl.GenericFloatProperty
+import com.bumble.appyx.interactions.core.ui.property.impl.Position
+import com.bumble.appyx.interactions.core.ui.property.impl.RotationX
+import com.bumble.appyx.interactions.core.ui.property.impl.Scale
+import com.bumble.appyx.interactions.core.ui.property.impl.ZIndex
+import com.bumble.appyx.interactions.core.ui.state.MatchedTargetUiState
+import com.bumble.appyx.mapState
+import com.bumble.appyx.transitionmodel.BaseMotionController
+
+class WidgetsStack3D<InteractionTarget : Any>(
+    uiContext: UiContext,
+) : BaseMotionController<InteractionTarget, SpotlightModel.State<InteractionTarget>, MutableUiState, TargetUiState>(
+    uiContext = uiContext,
+) {
+    private val width: Dp = uiContext.transitionBounds.widthDp
+    private val height: Dp = uiContext.transitionBounds.heightDp
+
+    private val scrollY = GenericFloatProperty(uiContext, GenericFloatProperty.Target(0f))
+    override val viewpointDimensions: List<Pair<(SpotlightModel.State<InteractionTarget>) -> Float, GenericFloatProperty>> =
+        listOf(
+            { state: SpotlightModel.State<InteractionTarget> -> state.activeIndex } to scrollY
+        )
+
+    private val created: TargetUiState = TargetUiState(
+        rotationX = RotationX.Target(0f),
+        position = Position.Target(DpOffset(0.dp, height)),
+        scale = Scale.Target(2.5f),
+        alpha = Alpha.Target(0f),
+        zIndex = ZIndex.Target(0f),
+    )
+
+    private val standard: TargetUiState = TargetUiState(
+        rotationX = RotationX.Target(0f),
+        position = Position.Target(DpOffset.Zero),
+        scale = Scale.Target(1f),
+        alpha = Alpha.Target(1f),
+        zIndex = ZIndex.Target(0f),
+    )
+
+    private val destroyed: TargetUiState = TargetUiState(
+        rotationX = RotationX.Target(0f),
+        position = Position.Target(DpOffset(0.dp, -height)),
+        scale = Scale.Target(0.25f),
+        alpha = Alpha.Target(0f),
+        zIndex = ZIndex.Target(0f),
+    )
```

Comment:
I suppose these don't affect the rendering since the widget doesn't trigger Spotlight's element state change operations, only scroll change, and then most of the UI is functional in `TargetUiState::toMutableUiState`? In that case we could just remove this as a red herring.

Reply:
You're right. I'll just leave a `default` one inside `private companion object` as we still need it for `toUiTargets` method.